archive/100-theorems-list/81_sum_of_prime_reciprocals_diverges.lean

```diff
@@ -0,0 +1,295 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import data.nat.prime
+import data.nat.sqrt
+import data.real.basic
+import data.finset
+import order.filter
+import topology.basic
+import algebra.big_operators.ring
+import topology.instances.ennreal
+import tactic.ring_exp
+import algebra.squarefree
+
+/-!
+# Divergence of the Prime Reciprocal Series
+This file proves Theorem 81 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+The theorem states that the sum of the reciprocals of all prime numbers diverges.
+The formalization follows Erdős's proof by upper and lower estimates.
+
+## References
+https://en.wikipedia.org/wiki/Divergence_of_the_sum_of_the_reciprocals_of_the_primes
+-/
+
+open_locale big_operators
+open_locale classical
+open nat pnat filter finset
+
+lemma card_le_div_nat {n p : ℕ} {Np : finset ℕ} (hp : 0 < p)
+  (hN : Np = {e ∈ range n | p ∣ (e + 1)}) : card Np ≤ n / p :=
```

Comment:
Is it normal to do things this way as opposed to just writing `card {e ∈ range n | p ∣ (e + 1)} ≤ n / p`? 
(This applies to some of the other lemmas as well)